rfc/github-org-migration.md

```diff
@@ -0,0 +1,415 @@
+# Github Org Migration
+
+![status:draft](https://img.shields.io/badge/status-draft-yellow.svg?style=flat-square)
+
+## Editors
+
+- [João Andrade](https://github.com/joaosa), [Protocol Labs](https://protocol.ai/)
+
+## Authors
+
+- [João Andrade](https://github.com/joaosa), [Protocol Labs](https://protocol.ai/)
+
+## Abstract
+
+Formalize a process for merging the [Filecoin Saturn](https://github.com/filecoin-saturn), [NFT.Storage](https://github.com/nftstorage), and [Web3 Storage](https://github.com/web3-storage) orgs into [w3s](https://github.com/w3s-project).
+
+Ensure a common understanding on how repositories should be migrated on a case-by-case basis.
+
+Provides recommendations and fixes to make this migration process as smooth and bulletproof for everyone.
+
+## Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+Migration is defined as the act of transferring ownership from either the [Filecoin Saturn](https://github.com/filecoin-saturn) or [Web3 Storage](https://github.com/web3-storage) orgs into [w3s](https://github.com/w3s-project) and ensuring all dependent systems continue operating transparently. If such a goal is not possible, then the necessary changes to either prevent the system from stopping its operation or recover it will be proposed.
```

Comment:
For context, one of the hardest things for me when grokking the IPFS/PL ecosystem is that seemingly the majority of times you click a documentation link to a github repo, it redirects you to some repo that is archived because the repo got passed around between a couple teams or something. If people had just left the public repos alone, those links would still link to repos with coherent docs and repo state, just maybe not committed to in awhile, but without the frustration of 'oh well i was eager to use this but now that it's explicitly abandoned I'm going to reevaluate things'

Reply:
@gobengo you make a very good point which I had not considered yet. I have experienced that confusion myself, so let's avoid it when we can. thank you!

In the case of public repos, I suppose forking and marking the "old" version as archived while linking to the new repo, for reference and maintenance reasons, should do it. As you said, there are no cost savings to be had here and this makes navigation clearer for end users. Any new changes will land in the fork.

For private repos (there are few sensitive ones on Saturn's side at least), I would still transfer them, as repo dependencies are solely defined internally and this would reduce the amount of artifacts we leave behind.
Also wrt to cost savings, given that "unless you are on a free plan, adding an outside collaborator to a private repository will use one of your paid licenses." ([ref](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/managing-outside-collaborators/adding-outside-collaborators-to-repositories-in-your-organization)), it makes sense to consolidate them into a single final org, so as to minimize the number of paid seats across the four orgs we will inherit in terms of maintenance (ideally, only one of them will need to evolve over time).

what do you think?

Reply:
> In the case of public-facing repos, I suppose forking and marking the "old" version as archived while linking to the new repo, for reference and maintenance reasons, should do it.

For https://github.com/web3-storage/w3up
If we can get to the point where the web3-storage org is on a free github plan, is there any reason at all there needs to be a w3s-project/w3up repo?

IMO we don't even need to have a new repo for w3up in a new org, so we wouldn't even need to do the 'marking the "old" version as archived' in that case. (I could be wrong, I just don't understand why w3up inherently needs to be migrated to a new github org when it could just stay put).

Reply:
> If we can get to the point where the web3-storage org is on a free github plan, is there any reason at all there needs to be a w3s-project/w3up repo?

Given that, as I see it, the primary goal here is enable cost savings (I would say a secondary goal is consolidation while minimizing complexity), it all comes down to the set of Github features we need to operate efficiently. 

The web3-storage org is using Github Enterprise and if we can do away with all the features listed [here](https://docs.github.com/en/get-started/learning-about-github/githubs-plans#github-team) and with no private repos, then it could be downgraded to a free org. That would enable us to leave things as they are.
I can't really tell what the nft.storage org is using as I'm not an org owner, but Saturn is using Github Teams.

Maybe we should have this discussion first, so as to figure out where to head next.
If we need any of those features then we'll need to move things around, as we'll need Github Teams and we only wanted to billed for that in a single org. 

I don't think we need Github Enterprise for web3-storage, but I'll do my best to verify that.
I'll also try to analyze and come up with a more concrete rationale on which features we need based on usage, but for now here's the list of Github Teams features:

- [ ] GitHub Support via email
  - This will likely need to exist in the org(s) we actually use. Have you all ever used this feature?
- [ ] 3,000 GitHub Actions minutes per month
  - Will need to verify this a bit deeper wrt to usage
- [ ] 2 GB GitHub Packages storage
  - same
- [ ] Advanced tools and insights in private repositories:
  - likely not need
- [ ] Required pull request reviewers
  - needs to be verified
- [ ] Multiple pull request reviewers
  - this is needed
- [ ] Draft pull requests
  - useful but maybe not needed
- [ ] Team pull request reviewers
  - needs to be verified
- [ ] Protected branches
  - this is needed
- [ ] Code owners
  - needs to be verified
- [ ] Scheduled reminders
  - needs to be verified
- [ ] GitHub Pages (private)
  - not needed
- [ ] Wikis
  - needs to be verified
- [ ] Repository insights graphs: Pulse, contributors, traffic, commits, code frequency, network, and forks
  - nice to have?
- [ ] The option to enable or disable GitHub Codespaces
  - needs to be verified

Reply:
Another argument in favor of a migration is the fact we can't add issues across orgs to project tracking. If we were to keep things as they are, we'd need to have a layer of project-specific issues on project tracking referring to the concrete ones in the final repos.

Reply:
@joaosa I don't know the backstory on why we use the more expensive GitHub Enterprise. We do make heavy-ish use of github actions.

>  That would be a decent win, but that would not solve project tracking 

One thing I'm sort of sad to lose that we have in web3-storage org but not w3s-project org is https://docs.github.com/en/issues/managing-your-tasks-with-tasklists/about-tasklists